### ISSUE TITLE
IInstaller to support IKernel and DebuggerEvaluator

### ISSUE THREAD

**Issue by sadasant** [microsoft/vscode-jupyter#10642]
- From: #10638
- After: #10641

Update `IInstaller` to use `IKernel` and the `DebuggerSession` depending on the environment.  Move the code from `dataViewerDependencyService` into the `IInstaller` to provide a universal installer experience that could be used to install any package with either a Kernel or the active debugger.

**Comment by DonJayamanne** [comment]
Closing this in favor of https://github.com/microsoft/vscode-jupyter/issues/10638
